schemes/gravity_wave_drag/gravity_wave_drag_frontogenesis.F90

```diff
@@ -0,0 +1,761 @@
+    real(kind_phys),    intent(out)   :: vtgw(:, :)               ! Meridional wind tendency from gravity waves [m s-2]
+    real(kind_phys),    intent(out)   :: ttgw(:, :)               ! Temperature tendency from gravity waves [K s-1]
+    real(kind_phys),    intent(out)   :: qtgw(:, :, :)            ! Constituent tendencies from gravity waves [kg kg-1 s-1]
+    real(kind_phys),    intent(out)   :: dttdf(:, :)              ! Temperature tendency from diffusion [K s-1]
+    real(kind_phys),    intent(out)   :: dttke(:, :)              ! Temperature tendency from kinetic energy dissipation [K s-1]
+
+    ! Copies of taucd in each direction for diagnostic.
+    real(kind_phys),    intent(out)   :: taucd_west(:, :)         ! Reynolds stress for waves in W direction, interfaces [N m-2]
+    real(kind_phys),    intent(out)   :: taucd_east(:, :)         ! Reynolds stress for waves in E direction, interfaces [N m-2]
+    real(kind_phys),    intent(out)   :: taucd_south(:, :)        ! Reynolds stress for waves in S direction, interfaces [N m-2]
+    real(kind_phys),    intent(out)   :: taucd_north(:, :)        ! Reynolds stress for waves in N direction, interfaces [N m-2]
+
+    ! Wind tendencies broken across five spectral bins for diagnostic.
+    real(kind_phys),    intent(out)   :: utend1(:, :)             ! U tendency c < -40 [m s-2]
+    real(kind_phys),    intent(out)   :: utend2(:, :)             ! U tendency -40 < c < -15 [m s-2]
+    real(kind_phys),    intent(out)   :: utend3(:, :)             ! U tendency -15 < c < 15 [m s-2]
+    real(kind_phys),    intent(out)   :: utend4(:, :)             ! U tendency 15 < c < 40 [m s-2]
+    real(kind_phys),    intent(out)   :: utend5(:, :)             ! U tendency c > 40 [m s-2]
+
+    character(len=512), intent(out)   :: errmsg
+    integer,            intent(out)   :: errflg
+
+    ! Local variables
+    integer :: i, k, m, l
+
+    ! Wavenumber fields
+    real(kind_phys) :: tau(ncol, -band_mid%ngwv:band_mid%ngwv, pver + 1)
+    real(kind_phys) :: gwut(ncol, pver, -band_mid%ngwv:band_mid%ngwv)
+    real(kind_phys) :: phase_speeds(ncol, -band_mid%ngwv:band_mid%ngwv)
+
+    real(kind_phys) :: utend(ncol, pver, 5)
+    integer         :: ix(ncol, -band_mid%ngwv:band_mid%ngwv)
+
+    real(kind_phys) :: egwdffi(ncol, pver+1)
+    real(kind_phys) :: effgw(ncol)
+    real(kind_phys) :: taucd(ncol, pver+1, 4)
+    real(kind_phys) :: um_flux(ncol)
+    real(kind_phys) :: vm_flux(ncol)
+    real(kind_phys) :: de(ncol)
+
+    errmsg = ''
+    errflg = 0
+
+    ! Efficiency of gravity wave momentum transfer
+    effgw = effgw_cm
+
+    ! Frontogenesis is too high at the poles (at least for the FV dycore), so introduce a polar taper
+    if (gw_polar_taper) then
+      effgw = effgw*cos(lat(:ncol))
+    endif
+
+    call gw_cm_src(ncol, pver, band_mid, &
+                   cm_desc, &
+                   u, v, frontgf(:ncol, :), &
```

Comment:
another repeat of this comment! I think all the `:ncol`, etc subsetting can be removed from this module.